### PR TITLE
ci: add ilab model evaluate mmlu to basic-workflow-tests

### PR DESCRIPTION
**Pull request by alimaredia**
**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.

### CODE REVIEW

File: scripts/basic-workflow-tests.sh
Comment (leseb) [inline scripts/basic-workflow-tests.sh:222]: task Test Evaluate

File: scripts/basic-workflow-tests.sh
Comment (leseb) [inline scripts/basic-workflow-tests.sh:170]: Should we add that step in the `test_download` function?

Comment (russellb) [reply scripts/basic-workflow-tests.sh:170]: yeah, makes sense to me

**Review by russellb — CHANGES_REQUESTED** [review]
Can you add a new command line arg that only runs this new functionality if enabled? On our smaller environments, I'm pretty sure this has to be off, right?

**Comment by nathan-weinberg** [comment]
> On our smaller environments, I'm pretty sure this has to be off, right?

Correct, this can't run on the standard GitHub runner we use AFAIK

File: scripts/basic-workflow-tests.sh
Comment (nathan-weinberg) [inline scripts/basic-workflow-tests.sh:169]: Per the Shellchecker
```suggestion
    export INSTRUCTLAB_EVAL_MMLU_MIN_TASKS=true 
    export HF_DATASETS_TRUST_REMOTE_CODE=true 
```

File: scripts/basic-workflow-tests.sh
Comment (russellb) [reply scripts/basic-workflow-tests.sh:170]: Actually, wouldn't the user workflow be to run this against the output of `ilab model train` ?

Comment (nathan-weinberg) [reply scripts/basic-workflow-tests.sh]: In theory, this can be used to evaluate any model (checkpoint or otherwise) - so it boils down to what we want to test here.

**Comment by russellb** [comment]
Note that the location this was added gets skipped in all of the workflows right now.  This PR will fix that, though: https://github.com/instructlab/instructlab/pull/1580

File: scripts/basic-workflow-tests.sh
Comment (danmcp) [inline scripts/basic-workflow-tests.sh:169]: Should this be -eq 1?

File: scripts/basic-workflow-tests.sh
Comment (danmcp) [inline scripts/basic-workflow-tests.sh:223]: This seems duplicated in test_evaluate

File: scripts/basic-workflow-tests.sh
Comment (danmcp) [inline scripts/basic-workflow-tests.sh:224]: Didn't this need to be moved up to take effect?

**Review by danmcp — CHANGES_REQUESTED** [review]

File: scripts/basic-workflow-tests.sh
Comment (russellb) [inline scripts/basic-workflow-tests.sh:172]: Could you change it to test the output of `ilab model train` ? It seems like that would be more representative of an end-to-end user flow, right?

File: scripts/basic-workflow-tests.sh
Comment (russellb) [reply scripts/basic-workflow-tests.sh:224]: thanks for remembering this comment from eariler, but it's good now as of https://github.com/instructlab/instructlab/pull/1580

It used to always exit before here. Now it will make it this far as long as `-f` (full training) is used. That does mean that `-e` without `-f` will not work. It's probably worth a check for that. I've been adding similar checks to `set_defaults`.

**Review by russellb — CHANGES_REQUESTED** [review]

File: scripts/basic-workflow-tests.sh
Comment (alimaredia) [reply scripts/basic-workflow-tests.sh:172]: Do you want that in this function or in `test_exec()` where the full workflow happens?

File: scripts/basic-workflow-tests.sh
Comment (russellb) [inline scripts/basic-workflow-tests.sh:96]: perhaps conditional on whether eval is turned on?

File: scripts/basic-workflow-tests.sh
Comment (russellb) [reply scripts/basic-workflow-tests.sh:203]: in here

**Comment by russellb** [comment]
I started a test of this change over here https://github.com/russellb/ilab-runner/actions/workflows/e2e-nvidia-a10g-x1-pr-1573.yml

**Comment by russellb** [comment]
> I started a test of this change over here https://github.com/russellb/ilab-runner/actions/workflows/e2e-nvidia-a10g-x1-pr-1573.yml

this passed!

I'm OK if you want to just file an issue on my larger comment (testing the output of train instead of a model we download). I don't want to block on that.

File: scripts/basic-workflow-tests.sh
Comment (russellb) [reply scripts/basic-workflow-tests.sh:100]: I pushed a commit to fix this

**Review by russellb — DISMISSED** [review]
I fixed the script issue that I was blocking on and added the doc update to reflect which job now tests eval.

I'm good with this (assuming a follow-up issue to look into testing the model that comes out of train)

**Comment by alimaredia** [comment]
@russellb The evaluation I'm adding here is post training stand alone model evaluation. 

The intercheckpoint evaluation I think you're talking about in your comments about "testing the output of `ilab model train`" happens within the training code by the eval library when the `--e2e` flag is added to `ilab model train`. That's something James is actively working on and isn't merged yet (https://github.com/instructlab/instructlab/pull/1466).

I don't think we'll need a follow up issue then right?

**Comment by alimaredia** [comment]
Oh I just realized you're probably talking about testing the final model output of the `ilab model train` command? 

If that's the case, I think we hold off on that until the e2e PR is merged so that we know where that model will be and what it will be called. Right now `ilab model train` is still spitting out a `.gguf` model right? We can't even run MMLU on that.

**Comment by russellb** [comment]
> Oh I just realized you're probably talking about testing the final model output of the `ilab model train` command?

Yes.

> If that's the case, I think we do file an issue and hold off on that until the training team can give us more information on where the new model will be and what it will be called. Right now `ilab model train` is still spitting out a `.gguf` model right? We can't even run MMLU on that.

OK ... but that's THE use case for eval. Running eval on a random model is not super useful.  What we need is eval of a model out of train. It's fine if that's still WIP though, that's why I want a follow-up issue

**Comment by danmcp** [comment]
What's the plan on getting the e2e to pass on this PR?

**Comment by russellb** [comment]
> What's the plan on getting the e2e to pass on this PR?

ouch, sorry, that's my fault. Thanks for pointing it out.

**Comment by alimaredia** [comment]
@danmcp Seeing as we got a clean run in for what's there I'd like to merge this PR. 

I've opened issues https://github.com/instructlab/instructlab/issues/1643 and https://github.com/instructlab/instructlab/issues/1642 to track the future work items on this PR and assigned @alinaryan and myself to them.

Is there anything else needed here before we can merge?

**Review by danmcp — APPROVED** [review]
Approving with the followup issues in place.  Thanks!

**Review by russellb — APPROVED** [review]

**Pull request merged**
